go_modules/lib/dependabot/go_modules/file_updater/go_mod_updater.rb

```diff
@@ -152,22 +151,27 @@ def run_go_vendor
         end
 
         def update_go_mod(dependencies)
-          deps = dependencies.map do |dep|
-            {
-              name: dep.name,
-              version: "v" + dep.version.sub(/^v/i, ""),
-              indirect: dep.requirements.empty?
-            }
+          tmp_go_file = "#{SecureRandom.hex}.go"
+
+          package = Dir.glob("[^\._]*.go").any? do |path|
+            !File.read(path).include?("// +build")
           end
 
-          body = SharedHelpers.run_helper_subprocess(
-            command: NativeHelpers.helper_path,
-            env: ENVIRONMENT,
-            function: "updateDependencyFile",
-            args: { dependencies: deps }
-          )
+          File.write(tmp_go_file, "package dummypkg\n") unless package
 
-          write_go_mod(body)
+          # TODO: go 1.18 will make `-d` the default behavior, so remove the flag then
+          command = +"go get -d"
+          # `go get` accepts multiple packages, each separated by a space
+          dependencies.each do |dep|
+            # Use version pinning rather than `latest` just in case
+            # a new version gets released in the middle of our run.
+            version = "v" + dep.version.sub(/^v/i, "")
+            command << " #{dep.name}@#{version}"
+          end
+          _, stderr, status = Open3.capture3(ENVIRONMENT, command)
+          handle_subprocess_error(stderr) unless status.success?
+        ensure
+          File.delete(tmp_go_file) if File.exist?(tmp_go_file)
```

Comment:
If we go this route, there's some DRY'ing up that could happen (maybe for the final call use `run_go_get()` with no args? I need to check how Ruby would handle this to ensure the generated command would simply be `go get -d`... this is purely my notes to self as I need to head to dinner+helping my wife with the kids.